docs/FlaskSQLAlchemy.md

```diff
@@ -0,0 +1,423 @@
+* The FLASK_ENV_CONFIG used in the the mentorship-backend are:
+  * prod
+  * test
+  * dev
+  * local
+  * stag
+
+* Among the various [configuration keys](https://flask-sqlalchemy.palletsprojects.com/en/2.x/config/) understood by the extension we take SQLALCHEMY_DATABASE_URI into consideration.
+
+**Note:** SQLALCHEMY_DATABASE_URI: The database URI that should be used for the connection. 
+URI or Uniform Resource Identifier is a string of characters that unambiguously identifies a particular resource.
+
+ * The SQLALCHEMY_DATABASE_URI for various FLASK_ENV_CONFIG are:
+
+   * [prod](https://github.com/anitab-org/mentorship-backend/blob/eae562267b1cbbb597ea3a8829b87ad373fa323c/config.py#L98)
+     ```
+     class ProductionConfig(BaseConfig):
+         """Production configuration."""
+
+         SQLALCHEMY_DATABASE_URI = BaseConfig.build_db_uri()
+         MOCK_EMAIL = False
+     ```
+
+   * [dev](https://github.com/anitab-org/mentorship-backend/blob/eae562267b1cbbb597ea3a8829b87ad373fa323c/config.py#L105)
+     ```
+     class DevelopmentConfig(BaseConfig):
+         """Development configuration."""
+
+         DEBUG = True
+         SQLALCHEMY_DATABASE_URI = BaseConfig.build_db_uri()
+     ```
+
+   * [stag](https://github.com/anitab-org/mentorship-backend/blob/eae562267b1cbbb597ea3a8829b87ad373fa323c/config.py#L112)
+     ```
+     class StagingConfig(BaseConfig):
+         """Staging configuration."""
+
+         DEBUG = True
+         SQLALCHEMY_DATABASE_URI = BaseConfig.build_db_uri()
+         MOCK_EMAIL = False
+     ```
+
+**Note:** For the above FLASK_ENV_CONFIG there is a function build_db_uri associated which can be found [here](https://github.com/anitab-org/mentorship-backend/blob/eae562267b1cbbb597ea3a8829b87ad373fa323c/config.py#L80), this is to build remote database URI using specific enviornment variables
+
+   * [local](https://github.com/anitab-org/mentorship-backend/blob/eae562267b1cbbb597ea3a8829b87ad373fa323c/config.py#L119)
+     ```
+     class LocalConfig(BaseConfig):
+         """Local configuration."""
+
+         DEBUG = True
+
+         # Using a local sqlite database
+         SQLALCHEMY_DATABASE_URI = "sqlite:///local_data.db"
+     ```
+
+   * [test](https://github.com/anitab-org/mentorship-backend/blob/eae562267b1cbbb597ea3a8829b87ad373fa323c/config.py#L128)
+     ```
+     class TestingConfig(BaseConfig):
+         """Testing configuration."""
+
+         TESTING = True
+         MOCK_EMAIL = True
+
+         # Use in-memory SQLite database for testing
+         SQLALCHEMY_DATABASE_URI = "sqlite://"
+     ```
+
+* To create all tables  
+  ```
+  def create_tables():
+      from app.database.sqlalchemy_extension import db
+
+      db.create_all()
+  ```
+
+* You can run the app by using the command
+  ```
+  python run.py
+  ```
+  Or
+  ```
+  export FLASK_APP=run.py
+  flask run
+  ```
+
+* Navigate to [http://127.0.0.1:5000/](http://127.0.0.1:5000/) in your browser
+
+  **Note:** Press CTRL+C to quit
+
+
+
```

Comment:
Why so many blank lines here??

Reply:
I didn't notice it in the doc should I resolve it

Reply:
?

Reply:
Yes, you should

Reply:
Ok I will